app-android/src/main/java/com/ustadmobile/port/android/view/PersonEditFragment.kt

```diff
@@ -173,94 +77,161 @@ class PersonEditFragment: UstadEditFragment<PersonWithAccount>(), PersonEditView
+            )
+
+            setContent {
+                MdcTheme {
+                    PersonEditScreen(viewModel)
+                }
+            }
         }
     }
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
-                              savedInstanceState: Bundle?): View {
-        val rootView: View
-        mBinding = FragmentPersonEditBinding.inflate(inflater, container, false).also {
-            rootView = it.root
-            it.imageViewLifecycleObserver = imageViewLifecycleObserver
-            it.hideClazzes =  arguments?.getString(ARG_HIDE_CLAZZES)?.toBoolean() ?: false
-        }
-
-        mPresenter = PersonEditPresenter(requireContext(), arguments.toStringMap(), this,
-                di, viewLifecycleOwner).withViewLifecycle()
-
-        mBinding?.usernameText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.usernameError = null
-        })
 
-        mBinding?.firstnamesText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.firstNamesError = null
-        })
+    companion object {
 
-        mBinding?.lastnameText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.lastNameError = null
-        })
-
-        mBinding?.genderValue?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.genderFieldError = null
-        })
-
-        mBinding?.passwordText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.passwordError = null
-        })
-
-        mBinding?.confirmPasswordText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.passwordConfirmError = null
-        })
-
-        mBinding?.birthdayText?.addTextChangedListener(ClearErrorTextWatcher {
-            mBinding?.dateOfBirthFieldError = null
-        })
-        mBinding?.emailText?.addTextChangedListener(ClearErrorTextWatcher{
-            mBinding?.emailError = null
-        })
-
-        mBinding?.usernameText?.filters = arrayOf(USERNAME_FILTER)
-        mBinding?.parentcontactText?.addTextChangedListener(RunAfterTextChangedTextWatcher {
-            parentContactError = null
-        })
-
-
-        return rootView
+        fun provideFactory(
+            di: DI,
+            owner: SavedStateRegistryOwner,
+            defaultArgs: Bundle? = null,
+        ): AbstractSavedStateViewModelFactory = object: AbstractSavedStateViewModelFactory(owner, defaultArgs) {
+            override fun <T : ViewModel?> create(
+                key: String,
+                modelClass: Class<T>,
+                handle: SavedStateHandle
+            ): T {
+                return PersonEditViewModel(di, SavedStateHandleAdapter(handle)) as T
+            }
+        }
     }
+}
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        mPresenter?.onCreate(backStackSavedState)
+@Composable
```

Comment:
Needs to add username field, password field, parental email field (as we had on the XML layout: https://github.com/UstadMobile/UstadMobile/blob/primary/app-android/src/main/res/layout/fragment_person_edit.xml#L399 ). You can add boolean fields on UiState for them to control if they are visible or not.